webapp/src/js/services/db-sync.js

```diff
@@ -64,7 +65,8 @@ angular
           heartbeat: 10000, // 10 seconds
           timeout: 1000 * 60 * 10, // 10 minutes
         },
-        allowed: () => $q.resolve(true)
+        allowed: () => $q.resolve(true),
+        onChange: RulesEngine.monitorExternalChanges,
```

Comment:
Would it be cleaner to have a db changes listener notify the RulesEngine when a Task doc is updated?

Reply:
I didn't want "DbSync" to know what a `task` is. It's RulesEngine's job to know what is relevant in its context.

Reply:
Sorry, that's not what I meant.

Instead of having an `onChange` listener in `DbSync` you could extend the Changes listener in the RulesEngine service to detect a change in a Task doc. That would mean the DbSync is decoupled from the RulesEngine altogether. Thinking about it more though it would probably be difficult to detect which changes originated remotely, and it might end up in an infinite loop...

I'm interested in continuing this discussion but I'm going to approve the PR to get AT rolling. We can refactor it later if the solution becomes clear!

Reply:
The loop is the reason I decided to listen to external changes. 

My original approach was to detect duplicate tasks when reading them: if the resulting tasks docs had duplicate, do exactly what the code does now (mark their requesters as dirty and recalculate and _then_ return the correct list of tasks). After a discussion with @SCdF , I changed it to this approach.